src/main/java/services/FormplayerLockRegistry.java

```diff
@@ -0,0 +1,112 @@
+package services;
+
+import exceptions.InterruptedRuntimeException;
+import io.sentry.event.Event;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.joda.time.DateTime;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.integration.support.locks.LockRegistry;
+import org.springframework.util.Assert;
+import util.Constants;
+import util.FormplayerSentry;
+
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.ReentrantLock;
+
+/**
+ * Created by willpride on 11/6/17.
+ */
+public class FormplayerLockRegistry implements LockRegistry {
+
+    @Autowired
+    private FormplayerSentry raven;
+
+    private final FormplayerReentrantLock[] lockTable;
+    // Lock objects for accessing the FormplayerReentrantLock table above
+    // These objects do not change
+    private final Object[] lockTableLocks;
+
+    private final int mask;
+
+    private final Log log = LogFactory.getLog(FormplayerLockRegistry.class);
+
+    public FormplayerLockRegistry() {
+        this(0xFF);
+    }
+
+    public FormplayerLockRegistry(int mask) {
+        String bits = Integer.toBinaryString(mask);
+        Assert.isTrue(bits.length() < 32 && (mask == 0 || bits.lastIndexOf('0') < bits.indexOf('1')), "Mask must be a power of 2 - 1");
+        this.mask = mask;
+        int arraySize = this.mask + 1;
+        this.lockTable = new FormplayerReentrantLock[arraySize];
+        this.lockTableLocks = new Object[arraySize];
+        for (int i = 0; i < arraySize; i++) {
+            this.lockTable[i] = new FormplayerReentrantLock();
+            this.lockTableLocks[i] = new Object();
+        }
+    }
+
+    private FormplayerReentrantLock setNewLock(Integer lockIndex) {
+        this.lockTable[lockIndex] = new FormplayerReentrantLock();
```

Comment:
Strictly speaking, this block should synchronize around the lock, in case it's called in ways you don't anticipate.